services/net/reporting/ReportingManager.cs

```diff
@@ -447,22 +453,20 @@ private async Task GenerateReportAsync(ReportRequestModel request, API.Areas.Ser
                 else
                     responseModel.FullTextFormatResponse = JsonDocument.Parse(JsonSerializer.Serialize(ex.Data["error"], _serializationOptions));
 
-                model.Status = ReportStatus.Failed;
-                model.Response = JsonDocument.Parse(JsonSerializer.Serialize(responseModel, _serializationOptions));
+                if (instanceModel != null)
+                {
+                    instanceModel.Status = ReportStatus.Failed;
+                    instanceModel.Response = JsonDocument.Parse(JsonSerializer.Serialize(responseModel, _serializationOptions));
+                }
             }
 
             if (instanceModel != null && request.GenerateInstance)
             {
                 // Update the report instance.
-                model.Id = instanceModel.Id;
-                model.Version = instanceModel.Version;
-                model.Content.ForEach(ric => ric.InstanceId = model.Id);
-                model.Subject = subject;
-                model.Body = fullTextFormatBody;
                 if (request.SendToSubscribers)
-                    instance.SentOn = instance.Status == ReportStatus.Accepted ? DateTime.UtcNow : null;
-                if (model.PublishedOn == null) model.PublishedOn = DateTime.UtcNow;
-                await this.Api.UpdateReportInstanceAsync(model);
+                    instanceModel.SentOn = instanceModel.Status == ReportStatus.Accepted ? DateTime.UtcNow : null;
```

Comment:
The original code looks like it has a few bugs in it.  I hope this update corrects them and doesn't introduce regression issues.